### PR TITLE
Make fetcher work

### PR DESCRIPTION
**Pull request by jeberl**
Summary
---------

- Edit permissions to allow fetcher to read+write s3 checkpoint and PDQ file
- Edit permissions to allow fetcher to get threat exchange api key secret
- create THREAT_EXCHANGE_PDQ_DATA_KEY environment variable

Test Plan
---------

`terraform apply`
Look at fetcher logs and see no errors
Delete privacy group checkpoints and pdq files using AWS UI and see that they are regenerated

### CODE REVIEW

File: hasher-matcher-actioner/terraform/fetcher/main.tf [hunk 1]
```diff
@@ -81,13 +83,12 @@ resource "aws_iam_role" "fetcher" {
...
-  statement {
-    effect    = "Allow"
-    actions   = ["s3:PutObject"]
-    resources = ["arn:aws:s3:::${var.threat_exchange_data.bucket_name}/${var.threat_exchange_data.pdq_data_file_key}"]
+    actions   = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:ListBucket",
+    ]
+    resources = ["*"]
```
Comment (eriksendc) [inline hasher-matcher-actioner/terraform/fetcher/main.tf:91]: I'm not an expert, but is `resources = ["*"]` an OK practice?

Comment (schatten) [reply hasher-matcher-actioner/terraform/fetcher/main.tf:91]: True, this could be scoped more narrowly.

**Review by schatten — APPROVED** [review]
Approving to unblock, but would love to see the s3 permissions constrained to (atleast) the bucket.

**Comment by BarrettOlson** [comment]
Now that we are connecting the fetcher e2e, similar to what @eriksendc, did with 'include_cloudfront_distribution' in `variables.tf` do you think we could add 'include_fetcher_trigger' variables? Could you also document somewhere what the fetcher does once deployed?

File: hasher-matcher-actioner/terraform/fetcher/main.tf (hunk 1 above)
Comment (eriksendc) [reply hasher-matcher-actioner/terraform/fetcher/main.tf:91]: fwiw I searched the existing terraform scripts and only found `resources = ["*"]` in things relating to logging in cloudfront.

Comment (jeberl) [reply hasher-matcher-actioner/terraform/fetcher/main.tf]: I tried a few different ways but the most specific I could get without failing seems to be to limit to s3. I've added this

**Pull request merged**